specutils/io/default_loaders/jwst_reader.py

```diff
@@ -22,52 +26,118 @@ def identify_jwst_fits(origin, *args, **kwargs):
+    ----------
+    filename: str
+        The path to the FITS file
+
+    Returns
+    -------
+    Spectrum1D
+        The spectrum contained in the file.
+    """
+    spectrum_list = _jwst_x1d_loader(filename, **kwargs)
+    if len(spectrum_list) == 1:
+        return spectrum_list[0]
+    else:
+        raise RuntimeError(f"{filename} has {len(spectrum_list)} spectra. "
+            "Use the SpectrumList.read.")
+
+
+@data_loader("JWST x1d multi", identifier=identify_jwst_x1d_fits, dtype=SpectrumList,
              extensions=['fits'])
-def jwst_loader(filename, spectral_axis_unit=None, **kwargs):
+def jwst_x1d_multi_loader(filename, **kwargs):
     """
-    Loader for JWST data files.
+    Loader for JWST x1d 1-D spectral data in FITS format
 
     Parameters
     ----------
-    file_name: str
+    filename: str
         The path to the FITS file
 
     Returns
     -------
-    data: SpectrumList
-        A list of the spectra that are contained in this file.
+    SpectrumList
+        A list of the spectra that are contained in the file.
+    """
+    return _jwst_x1d_loader(filename, **kwargs)
+
+
+def _jwst_x1d_loader(filename, **kwargs):
+    """Implementation of loader for JWST x1d 1-D spectral data in FITS format
+
+    Parameters
+    ----------
+    filename: str
+        The path to the FITS file
+
+    Returns
+    -------
+    SpectrumList
+        A list of the spectra that are contained in the file.
     """
 
     spectra = []
 
     with fits.open(filename) as hdulist:
+
         for hdu in hdulist:
+            # Read only the BinaryTableHDUs named EXTRACT1D
             if hdu.name != 'EXTRACT1D':
                 continue
 
-            # Provide reasonable defaults based on the units assigned by the
-            # extract1d step of the JWST pipeline. TUNIT fields should be
-            # populated by the pipeline, but it's apparently possible for them
-            # to be missing in some files.
-            wavelength_units = u.Unit(hdu.header.get('TUNIT1', 'um'))
-            flux_units = u.Unit(hdu.header.get('TUNIT2', 'mJy'))
-            error_units = u.Unit(hdu.header.get('TUNIT3', 'mJy'))
+            wavelength_units = u.Unit(hdu.columns["wavelength"].unit)
+            wavelength = hdu.data["wavelength"] * wavelength_units
+
+            # Determine if FLUX or SURF_BRIGHT column should be returned
+            # based on whether it is point or extended source
+            try:
+                srctype = hdu.header.get("srctype")
+            except KeyError:
+                # SRCTYPE is in primary header because there is only one spectrum
+                srctype = hdulist.header.get("srctype")
+
+            if srctype == "POINT":
+                flux_units = u.Unit(hdu.columns["flux"].unit)
+                flux = hdu.data["flux"] * flux_units
+
+                error_units = u.Unit(hdu.columns["error"].unit)
+                uncertainty = StdDevUncertainty(hdu.data["error"] * error_units)
+
```

Comment:
Having just sneaked into a discussion on the pros and cons of using the `Table` API for loading hdu data, I am wondering if this could be streamlined to
```
            data = Table.read(hdu)

            if srctype == "POINT":
                flux = Quantity(data["flux"])
                uncertainty = StdDevUncertainty(data["error"])
...
```
and so on for the other `srctype`s – see e.g. for comparison [muscles_sed.py](https://github.com/astropy/specutils/blob/master/specutils/io/default_loaders/muscles_sed.py).

Reply:
That looks good!  I'll give it a try.  Thanks for the tip.

Reply:
That worked very well.  Thanks.  Added in a new commit.  And I added some units checking to the tests, something I had overlooked before.

Reply:
Just remembered that `parsing_utils` has a function `spectrum_from_column_mapping` that in principle would even allow to simply specify this as dictionaries like
```
            if srctype == "POINT":
                column_mapping = {'wavelength': ('spectral_axis', None),
                                  'flux': ('flux', None),
                                  'error': ('uncertainty', None)}
            elif srctype == "EXTENDED":
                column_mapping = {'wavelength': ('spectral_axis', None),
                                  'surf_bright': ('flux', None),
                                  'sb_error': ('uncertainty', None)}
```
but this would require #573 to be merged to work without explicitly specifying the units - just for reference.

Reply:
I can add that simplification in a future PR.  👍